src/Admin/AbstractAdmin.php

```diff
@@ -605,16 +525,7 @@ public function getDataSourceIterator(): SourceIteratorInterface
         $fields = [];
 
         foreach ($this->getExportFields() as $key => $field) {
-            $label = $this->getTranslationLabel($field, 'export', 'label');
-            $transLabel = $this->trans($label);
-
-            // NEXT_MAJOR: Remove this hack, because all field labels will be translated with the major release
-            // No translation key exists
-            if ($transLabel === $label) {
-                $fields[$key] = $field;
-            } else {
-                $fields[$transLabel] = $field;
-            }
```

Comment:
Dont know either, a lot of tests are failing because of this change, maybe I didnt follow it right

Reply:
Before
```
[ 'valueToExport' ]
```
was creating the label `export.label_value_to_export`.

And if a translation for this label existed, we used it as header.

If a translation didn't exist we used the `key`. This allow to use it this way:
```
['mySpecialLabel' => 'valueToExport' ]
```
Only if `export.label_value_to_export` wasn't translated.

In the next major, they wanted to remove the `translated check` so 
```
['mySpecialLabel' => 'valueToExport' ]
```
will use `mySpecialLabel` even if a translation exist for `export.label_value_to_export`,

BUT:
- `[ 'valueToExport' ]` was convenient, and won't work
- there was 0 deprecation message for one way or another. So it needs an UPGRADE note, and adding a deprecation message on 3.x could be great.

We could improve the behavior:
If the key is an `int` then use `export.label_value_to_export`, if the `key` is a string use it.

The only issue we still have is: how/where do we translate the generated label `export.label_value_to_export` if we remove the `AbstractAdmin::trans` method ?

BTW there is some related doc in `action_export.rst`

Reply:
I just remembered that there is an issue for 4.x https://github.com/sonata-project/SonataAdminBundle/issues/5854

Reply:
@core23 You deprecated the `trans` method https://github.com/sonata-project/SonataAdminBundle/pull/4089,
Any solution here ?

Reply:
Wdyt about this...

I recover the translator injection on sonataadmin (only the setter)
I recover this behavior, but using $this->translator->trans, instead of the deprecated trans method

We discuss this on 3.x first to see the clear path for this.

I think this is the easier path to have this merged and keep discussing about the rest of NEXT_MAJOR changes that aren't clear to follow

Reply:
I agree.

Moreover IMHO the goal of the deprecation was to not promote the use of translator in the Admin class.
Having a private `translator` property and no public/protected `trans` method is going in this way.

Reply:
Done

Reply:
Can you make the translator property private ?

And maybe this could be a good idea to update the `NEXT_MAJOR: remove this hack` in the export method by adding something like `And find a way to translate the label somewhere else if we remove the translator in this class`.

Reply:
Should we do this on 3.x instead ? The comment change. The private thing should be done here yes

Reply:
Indeed, I opened the PR https://github.com/sonata-project/SonataAdminBundle/pull/6213